### PR TITLE
Add job to create a GitHub release

### PR DESCRIPTION
**Pull request by dlarocque**
Adds a job to the production release workflow to create a GitHub release.

I tested this on a minimal version (no build or npm publish) of the workflow on a fork. It adds and pushes tags the same way that the `scripts/release/release.ts` script does.
Test workflow: https://github.com/dlarocque/firebase-js-sdk/blob/master/.github/workflows/test-release.yml
Resulting release: https://github.com/dlarocque/firebase-js-sdk/releases/tag/firebase%405.5.5

This job assumes that the version packages PR was already merged.

### CODE REVIEW

File: .github/workflows/release-prod.yml [hunk 1]
```diff
@@ -124,3 +124,28 @@ jobs:
...
+    - name: Create Github release
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        # Get the newest release tag for the firebase package (e.g. firebase@10.12.0)
+        NEWEST_TAG=$(git describe --tags --match "firebase@[0-9]*.[0-9]*.[0-9]*" --abbrev=0)
+
+        # Get the release notes from the description of the most recent merged PR into the "release" branch
+        # See: https://github.com/firebase/firebase-js-sdk/pull/8236 for an example description
+        RELEASE_NOTES=$(gh pr list \
```
Comment (hsubox76) [inline .github/workflows/release-prod.yml:136]: Were we going to add in some text to say "For more detailed release notes, see [Firebase JavaScript SDK Release Notes](https://firebase.google.com/support/release-notes/js)."? You might want to test if you have to escape any of those markdown linking characters.

Changesets are better than commits but I think release notes may have more info than changesets.

File: .github/workflows/release-prod.yml [hunk 2]
```diff
@@ -124,3 +124,28 @@ jobs:
         curl -X POST -H "Content-Type:application/json" \
         -d "{\"version\":\"$BASE_VERSION\",\"date\":\"$DATE\"}" \
         $RELEASE_TRACKER_URL/logProduction
+    - name: Create Github release
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        # Get the newest release tag for the firebase package (e.g. firebase@10.12.0)
+        NEWEST_TAG=$(git describe --tags --match "firebase@[0-9]*.[0-9]*.[0-9]*" --abbrev=0)
```
Comment (hsubox76) [inline .github/workflows/release-prod.yml:132]: Nice use of --match to solve this problem.

File: .github/workflows/release-prod.yml (hunk 1 above)
Comment (dlarocque) [reply .github/workflows/release-prod.yml]: Done. Here's an example release with the new release notes: https://github.com/dlarocque/firebase-js-sdk/releases/tag/firebase%405.5.5

**Review by hsubox76 — APPROVED** [review]
Thanks for getting this done! Can't wait to see if it works on the next prod release.

**Pull request merged**